manager/integration/tests/test_node.py

```diff
@@ -1647,6 +1653,56 @@ def test_node_config_annotation_invalid(client, core_api,  # NOQA
     client.update(node, tags=[])
     wait_for_node_tag_update(client, node_name, ["storage"])
 
+    # Case7: Same disk name in annotation shouldn't intervene the node
+    # controller
+
+    # clean up any existing disk and create one disk for node
+    lht_hostId = get_self_host_id()
+    cleanup_node_disks(client, lht_hostId)
+    node = client.by_id_node(lht_hostId)
+    disks = node.disks
+    disk_path1 = create_host_disk(client, 'vol-disk-1',
+                                  str(Gi), lht_hostId)
+
+    # patch label and annotations to the node
+    core_api.patch_node(lht_hostId, {
+        "metadata": {
+            "labels": {
+                CREATE_DEFAULT_DISK_LABEL:
+                    CREATE_DEFAULT_DISK_LABEL_VALUE_CONFIG
+            },
+            "annotations": {
+                DEFAULT_DISK_CONFIG_ANNOTATION:
+                    '[{"path":"' + DEFAULT_DISK_PATH +
+                    '","allowScheduling":true,' +
+                    '"storageReserved": 1024,"tags": ["ssd", "fast"],' +
+                    '"name":"same-name"},' +
+                    '{"path":"' + disk_path1 +
+                    '","allowScheduling":true,' +
+                    '"storageReserved":1024,"name":"same-name"}]'
+            }
+        }
+    })
+
+    # same disk name shouldn't be applied to Longhorn.
+    time.sleep(NODE_UPDATE_WAIT_INTERVAL)
+    node = client.by_id_node(lht_hostId)
+    assert len(node.disks) == 0
+
+    # do cleanup.
```

Comment:
Don't need to do this. `reset_disk_and_tag_annotations` should has covered this already.